client/src/components/Cart/Cart.tsx

```diff
@@ -10,20 +10,24 @@ import {
 } from "@material-ui/core";
 import ItemCard from "src/components/ItemCard";
 import ItemCardCompact from "src/components/ItemCardCompact";
+import ItemPlaceholder from "src/components/ItemPlaceholder";
 
 function Cart({
   contents,
   collapse,
   updateItemQuantity,
 }: {
-  contents: CartItem[];
+  contents: (CartItem | null)[];
   collapse: boolean;
   updateItemQuantity?: (barcode: string, quantity: number) => void;
 }) {
   return (
     <div className="Cart">
       {contents.length > 0 &&
-        contents.map((cartItem) => {
+        contents.map((cartItem: CartItem | null, i: number) => {
+          if (!cartItem) {
+            return <ItemPlaceholder key={`item-${i}`} />;
+          }
```

Comment:
I think the loading logic and rendering logic should be 2 different components. 
They should be integrated in this manner:

```
{isLoading ?
<ShowLoadingCart length={6} /> : <Cart ...props />
```

Reply:
adopted model of separate list components for placholder vs real cards